app/lib/features/onboarding/pages/analytics_opt_in_page.dart

```diff
@@ -26,61 +34,280 @@ class AnalyticsOptInPage extends ConsumerWidget {
+            }, context);
+          },
+        ),
+        _buildCrashAnalyticsCard(
+          lang.sendCrashReportsTitle,
+          lang.sendCrashReportsInfo,
+          allowReportSending,
+          (value) async {
+            await setCanReportToSentry(value);
+            ref.invalidate(allowSentryReportingProvider);
+          },
+          context,
+          textTheme,
+        ),
+        _buildAnalyticsCard(
+          lang.basicTelemetry,
+          lang.basicTelemetryInfo,
+          basicTelemetryPref,
+          context,
+          textTheme,
+          ref,
+        ),
+        _buildCrashAnalyticsCard(
+          lang.appAnalytics,
+          lang.appAnalyticsInfo,
+          matomoAnalyticsEnabled,
+          (value) async {
+            await setMatomoAnalyticsEnabled(value);
+            ref.invalidate(allowMatomoAnalyticsProvider);
+          },
+          context,
+          textTheme,
+        ),
+        _buildAnalyticsCard(
+          lang.research,
+          lang.researchInfo,
+          researchPref,
+          context,
+          textTheme,
+          ref,
+        ),
+      ],
     );
   }
 
-  Widget _buildSkipActionButton(BuildContext context) {
-    return ActerPrimaryActionButton(
-      key: AnalyticsOptInPage.skipBtn,
-      onPressed: () => context.goNamed(Routes.main.name),
-      child: Text(
-        L10n.of(context).done,
-        style: Theme.of(context).textTheme.bodyMedium,
+  // Reusable widget for switch toggle option without card
+  Widget _buildSwitchOption(
+    String text,
+    bool value,
+    Future<void> Function() onToggle,
+    BuildContext context,
+  ) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      children: [
+        Text(text),
+        Transform.scale(
+          scale: 0.6,
+          child: Switch(value: value, onChanged: (_) => onToggle()),
+        ),
+      ],
+    );
+  }
+
+  // widget for switch toggle card
+  Widget _buildCrashAnalyticsCard(
+    String title,
+    String subtitle,
+    bool value,
+    Future<void> Function(bool) onToggle,
+    BuildContext context,
+    TextTheme textTheme,
+  ) {
+    return Card(
+      margin: const EdgeInsets.only(top: 10),
+      child: ListTile(
+        title: Text(title),
+        subtitle: Text(
+          subtitle,
+          style: textTheme.labelMedium?.copyWith(
+            color: Theme.of(context).colorScheme.onSurface,
+          ),
+        ),
+        trailing: Transform.scale(
+          scale: 0.6,
+          child: Switch(value: value, onChanged: onToggle),
+        ),
       ),
     );
   }
+
+  // widget for analytics card with separate preference
+  Widget _buildAnalyticsCard(
+    String title,
+    String subtitle,
+    String prefKey,
+    BuildContext context,
+    TextTheme textTheme,
+    WidgetRef ref,
+  ) {
+    return FutureBuilder<bool>(
+      future: _getAnalyticsPreference(prefKey),
+      builder: (context, snapshot) {
```

Comment:
same as above, let's please use a custom riverpod provider. Generally `FutureBuilder` isn't something that went well for us ever...